src/document.rs

```diff
@@ -106,6 +111,98 @@ fn value_to_string(value: &Value) -> String {
     }
 }
 
+/// Serializes a [`tv::DateTime`] object.
+///
+/// Since tantivy stores it as a single `i64` nanosecond timestamp, it is serialized and
+/// deserialized as one.
+fn serialize_datetime<S: Serializer>(
+    dt: &tv::DateTime,
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    dt.into_timestamp_nanos().serialize(serializer)
+}
+
+/// Deserializes a [`tv::DateTime`] object.
+///
+/// Since tantivy stores it as a single `i64` nanosecond timestamp, it is serialized and
+/// deserialized as one.
+fn deserialize_datetime<'de, D>(
+    deserializer: D,
+) -> Result<tv::DateTime, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    i64::deserialize(deserializer).map(tv::DateTime::from_timestamp_nanos)
+}
+
+/// An equivalent type to [`tantivy::schema::Value`] but uses tagging in its serialization to
```

Comment:
I am not sure I understand why this requires a different serialization schema compared to the upstream `Value` type.